### DISCUSSION TITLE
Password field in custom scripts

### DISCUSSION THREAD

**Discussion by ziggekatten** [netbox-community/netbox#6406] [Ideas]
I have some usecases for adding user/password data in the form, and it would be nice to have an option on the StringVar() object to treat it as a password field to avoid autocomplete and clear text when showing stuff. 

An 'hidden=True' as an option like label etc. 

What do you think?

**Comment by jeremystretch** [comment]
You can use Django's [PasswordInput widget](https://docs.djangoproject.com/en/3.2/ref/forms/widgets/#passwordinput) when declaring the script variable. This should work:

```
from django.forms import PasswordInput

password = StringVar(widget=PasswordInput)
```

**Reply by ziggekatten** [reply]
Never thought about that. 

Learning new things around Django every day:-)
